### DISCUSSION TITLE
Possible WMP 12 support in windows 10?

### DISCUSSION THREAD

**Discussion by ZachjuKamashi** [jojo2357/Music-Discord-Rich-Presence#79] [Ideas]
So I noticed WMP is supported but only on windows 11 and I believe it's due to the fact it was quite heavily changed. But I also believe it can be supported on Windows 10, because I noticed a comment in a reddit thread: https://www.reddit.com/r/discordapp/comments/8hmvp3/discord_rich_presence_for_windows_media_player_or/ And this makes me think it's very possible to have Rich presence in WMP 12, that is of course if the code for Live messenger is still within WMP 12 which it probably is.
![Screenshot (22)](https://user-images.githubusercontent.com/98941240/204525055-81986025-1ea1-4075-a83d-a4523275dc67.png)

**Comment by jojo2357** [comment]
I have a tool to test these things. it tells me (using the same interface that MDRP reads out of) that the windows 10 WMP does not report what it is playing to Windows Media Center. 

![image](https://user-images.githubusercontent.com/66704796/204592271-78ad20e1-ef19-4286-9bfd-60e6dd09a85e.png)

I am playing Johnny Cash in Groove and Imagine Dragons in WMP, but only one appeared. WMP shows up at the top because MDRP could tell that it was playing audio, but it couldnt figure out what that audio was.

The other giveaway is that for us windows 10 brethren, there should be a nice square with playing details when you change the volume.

For WMP:
![image](https://user-images.githubusercontent.com/66704796/204592837-b1817cc9-05e0-480c-aaa0-09c1b9cf4432.png)
Groove:
![image](https://user-images.githubusercontent.com/66704796/204592922-07d35483-9774-4a0f-b870-eb328b05d75a.png)

Not saying that it would be impossible, just that MDRP does not have that part of the architecture. If you can write software to figure out what WMP is playing, there is an API in MDRP that could be used so you don't have to rewrite the album art and other related architecture.
